docs/tableconfig_schema.rst

```diff
@@ -0,0 +1,172 @@
+
+Sample table config and descriptions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A sample table config is shown below which has sub-sections collasped. The sub sections will be described individually in following sections.
+
+The ``tableName`` should only contain alpha-numeric characters, hyphens ('-'), or underscores ('_'). Though using a double-underscore ('__') is not allowed and reserved for other features within Pinot.
+
+The ``tableType`` will indicate the type of the table, ``OFFLINE`` or ``REALTIME``. There are some settings specific to each type. This differentiation will be called out below as options are explained.
+
+.. code-block:: none
+
+    {
+      "tableName": "myPinotTable",
+      "tableType": "REALTIME"
+      "segmentsConfig": {},
+      "tableIndexConfig": {},
+      "tenants": {},
+      "routing": {},
+      "task": {},
+      "metadata": {}
+    }
+
+Segments Config Section
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``segmentsConfig`` section has information about configuring
+
+* Segment Retention - with the  ``retentionTimeUnit`` and ``retentionTimeValue`` options.
+* Segment Push - Using ``segmentPushFrequency`` to indicate how frequently segments are uploaded.
+* Replication - Using ``replication`` for offline tables and ``replicasPerPartition`` for realtime tables will indicate how many replicas of data will be present.
+* Schema - The name of the schema that's been uploaded to the controller
+* Time column - using ``timeColumnName`` and ``timeType``, this must match what's configured in the preceeding schema
+* Segment assignment strategy - Described more on the page `Customizing Pinot <customizations.html#segment-assignment-strategies>`_
+
+
+.. code-block:: none
+
+    "segmentsConfig": {
+      "retentionTimeUnit": "DAYS",
+      "retentionTimeValue": "5",
+      "segmentPushFrequency": "daily",
+      "segmentPushType": "APPEND",
+      "replication": "3",
+      "replicasPerPartition": "3",
+      "schemaName": "ugcGestureEvents",
+      "timeColumnName": "daysSinceEpoch",
+      "timeType": "DAYS",
+      "segmentAssignmentStrategy": "BalanceNumSegmentAssignmentStrategy"
+    },
+
+Table Index Config Section
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``tableIndexConfig`` section has information about how to configure:
+
+* Inverted Indexes - Using the ``invertedIndexColumns`` to specify a list of real column names as specified in the schema.
+* No Dictionary Columns - Using the ``noDictionaryColumns`` to specify a list of real column names as specified in the schema. Column names present will NOT have a dictionary created. More info on indexes can be found on the `Index Techniques <index_techniques.html>`_ page.
+* Sorted Column - Using the ``sortedColumn`` to specify a list of real column names as specified in the schema.
```

Comment:
Does this only affect realtime?

Reply:
I've confirmed this is only for realtime and updated the description.